chronos/queues/digests/processThreads.js

```diff
@@ -1,111 +1,140 @@
 // @flow
-const debug = require('debug')('chronos:queue:digest-process-threads');
 import { getTotalMessageCount, getNewMessageCount } from '../../models/message';
-import { getActiveThreadsInTimeframe } from '../../models/thread';
-import { MIN_TOTAL_MESSAGE_COUNT, MIN_NEW_MESSAGE_COUNT } from '../constants';
-import { getCommunityById } from '../../models/community';
-import { getChannelById } from '../../models/channel';
-import type { Timeframe, Thread, Threads, ThreadsWithData } from './types';
-
-export const getThreadsForDigest = async (timeframe: Timeframe) => {
-  // returns array of thread ids
-  const threadIds = await getActiveThreadsInTimeframe(timeframe);
-
-  // if no threadIds, escape
-  if (!threadIds || threadIds.length === 0) {
-    return null;
-  }
+import {
+  NEW_MESSAGE_COUNT_WEIGHT,
+  TOTAL_MESSAGE_COUNT_WEIGHT,
+} from '../constants';
+import { getCommunityById } from 'shared/db/queries/community';
+import { getChannelById } from 'shared/db/queries/channel';
+import type {
+  Timeframe,
+  ThreadWithDigestData,
+  CleanDigestThread,
+} from 'chronos/types';
+import { signCommunity } from 'shared/imgix';
+import truncate from 'shared/truncate';
 
-  // for each thread that was active in the last week, return a new array containing a record for each thread with the thread data and the message count
-  const messageCountPromises = threadIds.map(
-    async ({
-      communityId,
-      channelId,
-      id,
-      content,
-      messageCount,
-      ...thread
-    }) => ({
-      communityId,
-      channelId,
-      id,
-      title: content.title,
-      newMessageCount: await getNewMessageCount(id, timeframe),
-      totalMessageCount: messageCount || (await getTotalMessageCount(id)),
-    })
-  );
+import type { DBThread, DBCommunity, DBChannel } from 'shared/types';
+
+type DBThreadWithMetadata = {
+  ...$Exact<DBThread>,
+  community: DBCommunity,
+  channel: DBChannel,
+  messageCountString: string,
+  newMessageCount: number,
+  totalMessageCount: number,
+};
+
+type DBThreadWithMessageString = {
+  ...$Exact<DBThreadWithMetadata>,
+  messageCountString: string,
+};
 
-  const messageCounts = await Promise.all(messageCountPromises);
+export const attachMetadataToThreads = async (
+  threads: Array<DBThread>,
+  timeframe: Timeframe
+): Promise<Array<DBThreadWithMetadata>> => {
+  const promises = threads.map(
+    async (thread): Promise<DBThreadWithMessageString> => {
```

Comment:
Again, maybe serial instead of loading all of them in parallel?

Reply:
In this case it's just 3 queries though, and the first 2 will happen almost instantly, so maybe it's not too big of a problem?

Reply:
Yeah but it's 3 queries _per thread_, right? What I mean is running it for one thread, then the next, then the next, then the next,...

Or potentially running three batched queries for all the threads?

Reply:
I'm not sure how to batch the `getNewMessageCount` query, might need your help there

Reply:
Something like https://github.com/withspectrum/spectrum/pull/4358/commits/4a274c15396a79940315939d8304b9ae28ff6c02 @mxstbr ?